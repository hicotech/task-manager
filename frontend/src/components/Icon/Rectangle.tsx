import type { Icon } from '../../types/icon';

const Rectangle = ({
    size = '1.5rem',
    color = 'currentColor',
    'data-testid': dataTestId,
}: Icon) => (
    <svg
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 24 24"
        fill={color}
        width={size}
        height={size}
        data-testid={dataTestId}
    >
        <rect width={24} height={24} />
    </svg>
);

export default Rectangle;
