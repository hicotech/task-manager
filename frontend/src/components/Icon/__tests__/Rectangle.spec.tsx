import { render } from '@testing-library/react';
import type { Icon } from '../../../types/icon';
import Rectangle from '../Rectangle';

describe('Rectangle', () => {
    const testData: Required<Icon> = {
        size: '3.5rem',
        color: '#ff0000',
        'data-testid': 'rectangleIcon-testid',
    };

    describe('Renders Default', () => {
        const { container } = render(<Rectangle />);

        expect(container).toMatchSnapshot();
        expect(container.querySelector('svg > rect')).toBeInTheDocument();
    });

    describe('Props testing', () => {
        it('Default', () => {
            const { container } = render(<Rectangle />);

            expect(container.querySelector('svg')).toHaveAttribute('width', '1.5rem');
            expect(container.querySelector('svg')).toHaveAttribute('height', '1.5rem');
            expect(container.querySelector('svg')).toHaveAttribute('fill', 'currentColor');
        });

        it('Size', () => {
            const { container } = render(<Rectangle size={testData.size} />);

            expect(container).toMatchSnapshot();
            expect(container.querySelector('svg')).toHaveAttribute('width', testData.size);
            expect(container.querySelector('svg')).toHaveAttribute('height', testData.size);
        });

        it('Color', () => {
            const { container } = render(<Rectangle color={testData.color} />);

            expect(container).toMatchSnapshot();
            expect(container.querySelector('svg')).toHaveAttribute('fill', testData.color);
        });

        it('Data-testid', () => {
            const { getByTestId } = render(<Rectangle data-testid={testData['data-testid']} />);

            expect(getByTestId(testData['data-testid'])).toBeInTheDocument();
        });
    });
});
