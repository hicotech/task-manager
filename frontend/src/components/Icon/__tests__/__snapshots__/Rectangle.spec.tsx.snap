// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <svg
    fill="currentColor"
    height="1.5rem"
    viewBox="0 0 24 24"
    width="1.5rem"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="24"
      width="24"
    />
  </svg>
</div>
`;

exports[`Rectangle Props testing Color 1`] = `
<div>
  <svg
    fill="#ff0000"
    height="1.5rem"
    viewBox="0 0 24 24"
    width="1.5rem"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="24"
      width="24"
    />
  </svg>
</div>
`;

exports[`Rectangle Props testing Size 1`] = `
<div>
  <svg
    fill="currentColor"
    height="3.5rem"
    viewBox="0 0 24 24"
    width="3.5rem"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="24"
      width="24"
    />
  </svg>
</div>
`;
